core/src/main/java/org/infinispan/commands/write/DataWriteCommand.java

```diff
@@ -9,4 +10,31 @@
  * @since 4.0
  */
 public interface DataWriteCommand extends WriteCommand, DataCommand {
+
+   /**
+    * @return the {@link CommandInvocationId} associated to the command.
+    */
+   CommandInvocationId getCommandInvocationId();
+
+   /**
+    * Create the {@link BackupWriteCommand} to send to the backup owners.
+    * <p>
+    * The primary owner is the only member which creates the command to send it.
+    *
+    * @return the {@link BackupWriteCommand} to send to the backup owners.
+    */
+   default BackupWriteCommand createBackupWriteCommand() {
+      throw new UnsupportedOperationException();
+   }
+
+   /**
+    * Initializes the primary owner acknowledges with the return value and if it is successful or not.
+    *
+    * @param command     the {@link PrimaryAckCommand} to initialize.
+    * @param returnValue the local return value.
+    */
+   default void initPrimaryAck(PrimaryAckCommand command, Object returnValue) {
```

Comment:
actually I think it is a bad idea. the primary ack command needs the cache name and it is not available in the data write commands.